diff --git a/src/Cosi-BLS/block.lisp b/src/Cosi-BLS/block.lisp
index 423fdee..a200991 100644
--- a/src/Cosi-BLS/block.lisp
+++ b/src/Cosi-BLS/block.lisp
@@ -295,3 +295,10 @@ added to the blockchain."
       block
     (setf signature sig)
     (setf signature-bitmap bits)))
+
+
+
+
+(defun check-block-transactions (block)
+  (break)
+  t)
diff --git a/src/Cosi-BLS/cosi-handlers.lisp b/src/Cosi-BLS/cosi-handlers.lisp
index 52d6726..73dcbd7 100644
--- a/src/Cosi-BLS/cosi-handlers.lisp
+++ b/src/Cosi-BLS/cosi-handlers.lisp
@@ -701,7 +701,7 @@ Check that there are no forward references in spend position, then
 check that each TXIN and TXOUT is mathematically sound."
   (when *newtx-p*
     (return-from #1#
-      (cosi/proofs/newtx:check-block-transactions blk)))
+      (cosi/proofs:check-block-transactions blk)))
   (dolist (tx (get-block-transactions blk))
     (dolist (txin (trans-txins tx))
       (let ((key  (bev (txin-hashlock txin))))
diff --git a/src/Cosi-BLS/new-transactions.lisp b/src/Cosi-BLS/new-transactions.lisp
index 8c9dbc7..64b5d8d 100644
--- a/src/Cosi-BLS/new-transactions.lisp
+++ b/src/Cosi-BLS/new-transactions.lisp
@@ -1353,15 +1353,6 @@ ADDRESS here is taken to mean the same thing as the public key hash."
 
 
 
-(defun check-block-transactions (blk)
-  "Return nil if invalid block."
-  ;; It's unclear what this is supposed to do, under what
-  ;; circumstances this is called, etc. Talk to David and team.
-  (declare (ignore blk))                ; stub only for now! -mhd, 6/13/18
-  t)                                    ; tell caller everything's ok
-
-
-
 (defun clear-transactions-in-block-from-mempool (block)
   "Remove transactions that have just been added the block from the mempool that
    is globally bound to cosi-simgen:*mempool*. Note that these transactions may
diff --git a/src/Cosi-BLS/package.lisp b/src/Cosi-BLS/package.lisp
index 87767c3..2ccd0ea 100644
--- a/src/Cosi-BLS/package.lisp
+++ b/src/Cosi-BLS/package.lisp
@@ -195,6 +195,7 @@
    :block-signature-pkey
    :block-signature
    :block-merkle-root-hash
+   :check-block-transactions
 
    :*newtx-p*))                         ; for new transactions, short term temp! -mhd, 6/12/18
 
@@ -219,7 +220,6 @@
    initial-total-coin-amount
    validate-transaction
    get-transactions-for-new-block
-   check-block-transactions
    dump-tx
    dump-txs
    clear-transactions-in-block-from-mempool
